### ISSUE TITLE
sort-imports examples don't pass with default config

### ISSUE THREAD

**Issue by epmatsw** [eslint/eslint#8734]
**Tell us about your environment**

http://eslint.org/demo/ with ecmaVersion6 and sourceType module.

* **ESLint Version:**
* **Node Version:**
* **npm Version:**

**What parser (default, Babel-ESLint, etc.) are you using?**

**Please show your full configuration:**

The default configuration on http://eslint.org/demo/, but with ecmaVersion 6, sourceType of module, turned off `no-unused-var`, turned on `sort-imports`.

**What did you do? Please include the actual source code causing the issue.**

```js
import 'module-without-export.js';
import * as foo from 'foo.js';
import * as bar from 'bar.js';
import {alpha, beta} from 'alpha.js';
import {delta, gamma} from 'delta.js';
import a from 'baz.js';
import b from 'qux.js';
```

```js
import 'foo.js'
import * from 'bar.js';
import {a, b} from 'baz.js';
import c from 'qux.js';
```

**What did you expect to happen?**
Those are the two examples of "correct" default code from http://eslint.org/docs/rules/sort-imports#default-settings. I expected them to both pass without issue in the default `sort-imports` behavior.

**What actually happened? Please include the actual, raw output from ESLint.**

For the first one, it gives `3:1 - Imports should be sorted alphabetically. (sort-imports)` (this matches the issue I was investigating when I found this, unexpected complaints about sort order). For the second one, it gives `2:10 - Parsing error: Unexpected token from ()`, which is pretty strange.

**Comment by epmatsw** [comment]
Having now looked at it, it's cause `foo` is alphabetically after `bar` and because the `import * from` needs to be an `import * as <something> from`

**Comment by i-ron-y** [comment]
Fixed the doc, made a PR.
